src/message/fetchActions.js

```diff
@@ -249,7 +249,13 @@ export const doInitialFetch = () => async (dispatch: Dispatch, getState: GetStat
     return;
   }
 
-  dispatch(realmInit(initData, serverSettings.zulip_version));
+  dispatch(
+    realmInit(
+      initData,
+      serverSettings.zulip_feature_level === undefined ? null : serverSettings.zulip_feature_level,
```

Comment:
Instead of getting this from `serverSettings`, we can actually get the `zulip_feature_level` from `initData`. (Which also means this line and `realmInit` don't need to change, and the reducer can handle it.)

That's preferable because for recent servers we can also get `zulip_version` from there. We make this `getServerSettings` call here because older servers send `zulip_version` only there; but `zulip_feature_level` appears on the same set of servers in one place as the other place.

This isn't well documented, sorry -- not your fault that you didn't spot that. The initial data from `/register` is one of the couple of main areas we haven't yet covered in our API documentation.

Reply:
> Instead of getting this from `serverSettings`, we can actually get the `zulip_feature_level` from `initData`. (Which also means this line and `realmInit` don't need to change, and the reducer can handle it.)

Mm, nice! Perhaps the type `InitialDataRealm`, or even `InitialDataBase`, might be a place to add `zulip_feature_level?: number`, to make `initData` well-typed?